Undo Button adding, Detection test, Prevent wrong convertion …

greenlikeorange opened the pull request:
- Detection test is use MochaJS
- Undo button is now available
- Version is upgrade to 0.1.0 becuse there are some big changes
- Everything is perfectly work

Cheer.!

Commit:
Undo Button adding, Detection test, Prevent wrong convertion

- Detection test is use MochaJS
- Undo button is now available
- Version is upgrade to 0.1.0 becuse there are some big changes
- Everything is prefet and work

Commit:
Version Update

Commit:
Hot Fix on buttons tiggle

greenlikeorange committed:
Update README.md

swanhtet1992 commented on scripts/contentscript.js:
```diff
@@ -16,64 +16,133 @@
- *same with $(document).ready(function(){ });
- *Got it from http://youmightnotneedjquery.com/ 
- */
-function ready(fn) {
```
@greenlikeorange We don't need this? Could you please explain me so that I could decide.

greenlikeorange replied:
A, content script on browser are run when document is ready

https://developer.chrome.com/extensions/content_scripts

> Note: With "document_idle", content scripts may not necessarily receive the window.onload event, because they may run after it has already fired. In most cases, listening for the onload event is unnecessary for content scripts running at "document_idle" because they are guaranteed to run after the DOM is complete. If your script definitely needs to run after window.onload, you can check if onload has already fired by using the document.readyState property.

swanhtet1992 replied:
OK, got it.

swanhtet1992 commented on scripts/contentscript.js:
```diff
@@ -16,64 +16,133 @@
+
+        // document.getElementsByName('xhpc_message')[0].value =  textBox.value;
+        /* Instance of Finding "xhpc_message" we can do this little trick ;) */
+
```
Tha's great.

swanhtet1992 commented on scripts/contentscript.js:
```diff
@@ -16,64 +16,133 @@
+        /* Instance of Finding "xhpc_message" we can do this little trick ;) */
+
+        focusedTextBox.parentNode.removeChild(magicButton);
+        focusedTextBox.parentNode.appendChild(undoButton);
```
This one great.

swanhtet1992 commented on scripts/contentscript.js:
```diff
@@ -16,64 +16,133 @@
+    /* Just clone is enough */
+    undoButton = magicButton.cloneNode(true);
+    undoButton.innerHTML = "Undo";
+
```
I like your undo button idea.

swanhtet1992 commented:
Overall, this is awesome. Great work.

Cheers,
SH

Commit:
keyup event binding
===================

`keyup` event binding for prevnet **Undo** button showing when user is edit text

Commit:
Merge branch 'master' of github.com:greenlikeorange/MagicButton

swanhtet1992 commented on scripts/contentscript.js:
```diff
@@ -16,64 +16,144 @@
+
+      }, false);
+
+      // box.addEventListener("blur", function(event){
```
Do we need this? If we don't, I will remove it.

swanhtet1992 commented on test/detector_test.js:
```diff
@@ -0,0 +1,33 @@
+// var detector = require("../scripts/_detector.js");
```
If these comments are not needed, please remove it.

swanhtet1992 commented:
Merged. 👍👍👍👍👍
